libr/util/buf.c

```diff
@@ -223,7 +234,7 @@ R_API ut64 r_buf_size (RBuffer *b) {
 		}
 		return 0LL;
 	}
-	return b->empty? 0: R_MIN (b->limit, R_MAX (0, b->length - b->offset));
+	return b->empty? 0: minimumZeroMaximum (b->limit, b->length, b->offset);
```

Comment:
Can you just inline it `R_MIN (b->limit, b->offset < b->length ? b->length-b->offset : 0)` instead of inventing a new function?

Reply:
But that would mean code triplication, because I would need to inline it at three places.

Also, I hate the ternary operator.

Reply:
having a separate function makes the logic more clear and avoids duplication of inlined code. (the compiler can decide). But i dont think this change makes the code more readable because the function name is a bit confusing

Reply:
Alternatively, you can add a check `if (b->offset >= b->length) return 0; ` then you don't need the helper function at all because the logic is simple enough and can be inlined.

Reply:
> Alternatively, you can add a check if (b->offset >= b->length) return 0; then you don't need the helper function at all because the logic is simple enough and can be inlined.

That would still mean code duplication.